miniapp/CMakeLists.txt

```diff
@@ -8,80 +8,61 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-include(GNUInstallDirs)
-
 add_library(DLAF_miniapp INTERFACE)
 target_link_libraries(DLAF_miniapp INTERFACE DLAF pika::pika)
 target_include_directories(DLAF_miniapp
   INTERFACE
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
 )
 
-add_executable(miniapp_cholesky miniapp_cholesky.cpp)
-target_link_libraries(miniapp_cholesky PRIVATE dlaf.core dlaf.auxiliary dlaf.factorization)
-target_link_libraries(miniapp_cholesky PRIVATE DLAF_miniapp)
-target_add_warnings(miniapp_cholesky)
-DLAF_addPrecompiledHeaders(miniapp_cholesky)
-install(TARGETS miniapp_cholesky RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+DLAF_addMiniapp(miniapp_cholesky
+  SOURCES miniapp_cholesky.cpp
+  LIBRARIES dlaf.core dlaf.auxiliary dlaf.factorization
```

Comment:
Just realised that in this miniapp we still rely on "internal" targets, but by adding `DLAF_miniapp` we are implicitly depending on the full `DLAF` dependency.

Reply:
Good catch. I think `DLAF_miniapp` probably doesn't need all the individual libraries. Will update it to depend only on a subset if that's the case.

Reply:
Hrm, so `dlaf.core` would be enough for `DLAF_miniapp`. However, we use a mix in the miniapps. The one above depends on individual sub-libraries, but most others depend on `DLAF`.

Do we want to keep depending on individual libraries only? If yes, I'll change the ones depending on `DLAF` to more fine-grained dependencies.

Reply:
I think is better to link with DLAF, at least until we find a way to workaround the problem of dependencies of object libraries.

Reply:
Everything links to `DLAF` now. The miniapps link explicitly to `DLAF` even though they would get it through `DLAF_miniapp` as well. Is that ok with you?